docs/_posts/2017-09-26-react-v16.0.md

```diff
@@ -0,0 +1,133 @@
+
+That amounts to a combined **32% size decrease compared to the previous version (30% post-gzip)**.
+
+The size difference is partly attributable to a change in packaging. React now uses [Rollup](https://rollupjs.org/) to create flat bundles for each of its different target formats, resulting in both size and runtime performance wins. The 
+
+### New core architecture
+
+React 16 is the first version of React built on top of a new core architecture, codenamed "Fiber." You can read all about this project over on Facebook's engineering blog (TODO: link). (Spoiler: we rewrote React!)
+
+Most of the features in this release, like error boundaries and fragments, were made possible by the rewritten core. Over the next few releases, you can expect more new features as we begin to unlock the full potential of React.
+
+Perhaps the most exciting area we're working on is **async rendering**—a strategy for cooperatively scheduling rendering work by periodically yielding execution to the browser. The upshot is that, with async rendering, apps are more responsive because React avoids blocking the main thread.
+
+See this demo for an early peek at the types of problems async rendering can solve. TODO: link to demo.
+
+We think async rendering is a big deal, and represents the future of React. To make migration to v16.0 as smooth as possible, we're not enabling any async features yet, but we're excited to roll this out in the coming months. Stay tuned!
+
+## Upgrading
+
+Although React 16 includes significant internal changes, in terms of upgrading, you can think of this like any other major React release. We've been serving React 16 to Facebook and Messenger.com users since earlier this year, and we released several beta and release candidate versions to flush out additional issues. With minor exceptions, **if your app runs in 15.6 without any warnings, it should work in 16.**
+
+### Deprecations
+
+Hydrating a server-rendered container now has an explicit API. Use ReactDOM.hydrate instead of ReactDOM.render if you're reviving server-rendered HTML. Keep using ReactDOM.render if you're just doing client-side rendering.
+
+Documentation (TODO: link)
+
+### Breaking changes
+
+* React 15 had limited, undocumented support for error boundaries using `unstable_handleError`. This method has been renamed to `componentDidCatch`. You can use a codemod to [automatically migrate to the new API](https://github.com/reactjs/react-codemod#error-boundaries).
+* `ReactDOM.render` and `ReactDOM.unstable_renderIntoContainer` now return null if called from inside a lifecycle method. To work around this, you can use [portals](https://github.com/facebook/react/issues/10309#issuecomment-318433235) or [refs[(https://github.com/facebook/react/issues/10309#issuecomment-318434635).
+* `setState`:
+  * Calling `setState` with null no longer triggers an update. This allows you to decide in an updater function if you want to re-render.
```

Comment:
Does this also mean that _returning_ `null` from a `setState` callback (e.g. `this.setState(() => null)`) will avoid triggering an update?